src/wp-admin/includes/file.php

```diff
@@ -972,11 +975,16 @@ function wp_handle_upload_error( &$file, $message ) {
 		$type = '';
 	}
 
+	if ( isset( $overrides['upload_dir'] ) ) {
+		$uploads = $overrides['upload_dir'];
```

Comment:
Is it worth validating the shape here?

Reply:
Hm, perhaps, but thinking it may be better to let the developers handle it? I mean, this is new functionality and there is sufficient documentation (I hope) about what is expected and how to get it. Can add some "defensive" isset() and type checks, maybe, but that may prevent seeing errors while developing?

Reply:
Added some sanity checks as part of ensuring the uploads directory exists when plugins use custom data.